pkg/cloud/azure/services/networkinterfaces/networkinterfaces.go

```diff
@@ -120,6 +117,20 @@ func (s *Service) CreateOrUpdate(ctx context.Context, spec azure.Spec) error {
 	}
 	nicConfig.LoadBalancerBackendAddressPools = &backendAddressPools
 
+	if nicSpec.PublicIPName != "" {
```

Comment:
please reuse the current public load balancer and reuse the nat rule for bastion vm instead of control plane vm